libtiledbsoma/src/utils/arrow_adapter.cc

```diff
@@ -417,29 +417,122 @@ Dimension ArrowAdapter::_create_dim(
         case TILEDB_DATETIME_SEC:
         case TILEDB_DATETIME_MS:
         case TILEDB_DATETIME_US:
-        case TILEDB_DATETIME_NS:
-            return Dimension::create(
-                *ctx, name, type, (uint64_t*)buff, (uint64_t*)buff + 2);
-        case TILEDB_INT8:
+        case TILEDB_DATETIME_NS: {
+            // Sadly we cannot put this in the centralized _create_dim_aux
+            // in the header file. That's because we need utils/logger.h
+            // -- which is a _fixed_ relative path from _this_ .cc file
+            // but a _varying_ relative path from all the places that
+            // #include arrow_adapter.h. Hence the code duplication in
+            // logging statements. :(
+            uint64_t* b = (uint64_t*)buff;
+            LOG_DEBUG(fmt::format(
+                "_create_dim name={} b={} b1={} b2={}",
+                name,
+                b[0],
+                b[1],
+                b[2]));
```

Comment:
https://github.com/single-cell-data/TileDB-SOMA/blob/main/libtiledbsoma/src/soma/soma_array.h#L302-L307

In cases where we use string formatting in the header files, we avoid using `fmt::format` and make a TODO note to replace these lines with `std::Format` once we bump to C++20. I'm not saying to remove these lines in here; it is fine as is but just an fyi.